File: frontend/src/objects/review/ReviewAppealDecision.ts
// Business note: frontend mirror of a backend service object; keep names and fields aligned unless this file is explicitly frontend-only.
import type {
  AppealRole,
  ReasonText,
} from '@/objects/core/SharedObjects'

export type ReviewAppealDecision = {
  appellantRole: AppealRole
  reason: ReasonText
}

File: backend/src/main/scala/services/review/objects/ReviewAppealDecision.scala
package services.review.objects

// Business note: service-owned domain object; mirror it in the frontend only when it is part of the protocol surface.
import system.objects.given

import system.objects.*
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.*

final case class ReviewAppealDecision(
    appellantRole: AppealRole,
    reason: ReasonText,
)
object ReviewAppealDecision:
  given Encoder[ReviewAppealDecision] = deriveEncoder
  given Decoder[ReviewAppealDecision] = deriveDecoder
